quinn-proto/src/endpoint.rs

```diff
@@ -379,12 +379,21 @@ impl Endpoint {
                 .cloned()
         };
         if let Some(conn_id) = conn {
-            let conn = &mut self.connections[conn_id.0];
-            let was_handshake = conn.is_handshaking();
-            let remaining = conn.handle_decode(now, ecn, partial_decode);
-            if was_handshake && !conn.is_handshaking() && conn.side().is_server() {
-                self.incoming_handshakes -= 1;
-                self.ctx.incoming.push_back(conn_id);
+            let was_handshake = self.connections[conn_id.0].is_handshaking();
+            let remaining = self.connections[conn_id.0].handle_decode(now, ecn, partial_decode);
+            if was_handshake && !self.connections[conn_id.0].is_handshaking() {
+                // Newly established connection
+                if self.connections[conn_id.0].side().is_server() {
+                    self.incoming_handshakes -= 1;
+                    self.ctx.incoming.push_back(conn_id);
+                }
+                if self.config.local_cid_len != 0 {
+                    // endpoints SHOULD provide and maintain at least eight connection IDs
+                    for _ in 0..8 {
```

Comment:
Should we somehow document what this `8` is?

Reply:
Also it feels odd for the majority of this hunk to live in `Endpoint`, why is that?

Reply:
Most of the changes in this hunk are semantic no-ops to ensure we can call `self.new_cid()` without an outstanding borrow. That said, I think the placement is justified. We're doing two things here: reporting the establishment of a new incoming connection, and providing new connections with their alternate CIDs. Both of these inherently involve modifying `Endpoint` state, for the incoming connection queue and the connection ID table respectively.

I'll insert a constant for the CID count.